spec/controllers/admin/deduplicate_patients_controller_spec.rb

```diff
@@ -0,0 +1,18 @@
+require "rails_helper"
+
+RSpec.describe Admin::DeduplicatePatientsController, type: :controller do
+  context "#merge" do
+    it "Merges patients given their IDs" do
+      admin = create(:admin, :power_user)
+      sign_in(admin.email_authentication)
+
+      patients = [create(:patient, full_name: "Patient one"), create(:patient, full_name: "Patient two")]
+
+      post :merge, params: {duplicate_patients: patients.map(&:id)}
+
+      expect(Patient.count).to eq 1
+      expect(Patient.first.merged_by_user_id).to eq admin.id
+      expect(Patient.with_discarded.count).to eq 3
```

Comment:
Can we make some more surgical assertions here?

```ruby
expect(patients).to all(be_discarded)
expect(patients.map(&:merged_by_user_id).to all(eq(admin.id))
expect(Patient.count).to eq 1
```

Come to think of it, should the controller test just ensure that the deduplication service is invoked with the right arguments?

Reply:
Thanks, that helps! Fixed the assertions in d225eb056e9f139660794f031514b355e614a8fa.

We actually went the mocking route first, then decided this is simpler. The matcher for the argument expectations in the mock was getting hairy since we'd have to assert on active record relations, etc.

Reply:
Ah yeah, classic. You end up having to mock a whole series of methods - "please return this double instead of a real AR relation" etc. 🆒 beans.